### DISCUSSION TITLE
V3.5.2 - How to enable a menu bar?

### DISCUSSION THREAD

**Discussion by chrisjj** [sumatrapdfreader/sumatrapdf#4640] [Q&A]
First, thanks for the program!

My question. How do I enable a main menu bar? Currently (default install), main menu is under a hamburger button - to save space that does not need saving here, since I have an excess of tis which I would like to be used for a standard WIG menu bar.

No answer found in previous discussions here or Help.

Thanks.

**Comment by GitHubRulesOK** [comment]
The core viewer is Multi-format MuPDF which is one page frame and has no real GUI such as menu tabs scrolling bars etc.
![image](https://github.com/user-attachments/assets/55f034dc-26eb-416b-86c4-d1cfe5e45093)

SumatraPDF adds a menu and scrollbars to pages larger than a single frame
![image](https://github.com/user-attachments/assets/71e361cb-cdaa-4fea-a2de-90a470ad64e4)
and a single line toolbar
![image](https://github.com/user-attachments/assets/14353228-a94e-42b9-87d5-07f8445abb65)

![image](https://github.com/user-attachments/assets/cf6fc3df-c100-4147-8453-e6662a30f832)
Alternately (by default) can be tabbed or switch tabs OFF and restart in menu mode
![image](https://github.com/user-attachments/assets/f1d56b8e-7822-4388-832a-2d689a0a095e)

**Reply by chrisjj** [reply]
What is a single frame please?

**Reply by larrybud2004** [reply]
> What is a single frame please?

I think he means when there's only 1 document opened, but I could be wrong.

**Comment by GitHubRulesOK** [comment]
When fit a single page is active the placement can be any size and scrollbars detract from that working area.
Clearly not a  problem for users with modern large monitors but for mobile or legacy use the canvas area was designed to be single framed. 

Here shown in one of a few modern browsers with fit page active.

Admittedly now a rarer situation with mainstream browsers like Chrome or Firefox that dont support SumatraPDF as a plug-in. But many users still utilise SumatraPDF.EXE as a core viewer within their applications or smaller devices.

![image](https://github.com/user-attachments/assets/c01d57e7-e45a-42b3-a4f9-377f32c94c37)![image](https://github.com/user-attachments/assets/c92c3ad8-5504-42ec-9bbd-4ed6e26c3490)

**Comment by chrisjj** [comment]
Thanks, but still I do not know what is meant by single frame.
>
> Clearly not a problem for users with modern large monitors but for mobile
> or legacy use the canvas area was designed to be single framed.
>
So, the only option available for modern large monitors is one designed for
m9bike and legacy?

> Here shown in one of a few modern browsers with fit page active.
>
I am not using a browser. I am using the SumatraPDF exe.

**Comment by GitHubRulesOK** [comment]
Two alternative Windows light-weight viewers that behaved similar (no scrollbar in non continuous single frame) but have more GUI options are Okular so here my latest is fairly old (about version 2019:03/2019:04 so later should have more menus including sound and dark screen features and unknown if they added different scroll behaviour) and XDF latest 2024 shown is 4.05
![image](https://github.com/user-attachments/assets/01c1da68-fe8f-4907-9e55-14d016ebe65a)

Most of the simpler viewers are based on MuPDF or xPDF and thus have very similar core PDF rendering functions of no scroll in fit a single (non-continuous) page or have widely differing GUI enhancements around those framed engines. 
![image](https://github.com/user-attachments/assets/ba18fcce-7014-4d1d-b1ca-769cf34b0e2a)

**Comment by chrisjj** [comment]
Thanks but I was asking about just SumatraPDF.

**Comment by larrybud2004** [comment]
So are we saying that the menu bar cannot be displayed and we have to use the hamburger menu? There's an option in the settings called "ShowMenuBar" which has no effect on anything.

**Reply by GitHubRulesOK** [reply]
@larrybud2004 

Current SumatraPDF uses legacy menubar or more modern tabs with drop down  (so only 2 of 3 choices and the third mix is not normal so unsupported  in the releases)
![image](https://github.com/user-attachments/assets/4a3202cf-d885-42e4-ac4c-49302f84e45c)


![image](https://github.com/user-attachments/assets/508b1bfd-2bc6-46b2-ba63-9fa7a72611e3)

**Comment by larrybud2004** [comment]
How does one get the so-called "legacy menubar" to appear? 

"so only 2 of 3 choices and the third mix is not normal so unsupported in the releases"

I don't know what that means.

**Reply by GitHubRulesOK** [reply]
@larrybud2004 
switch off tabs and restart SumatraPDF the windows will be un-tabbed
so the mix of tabs and menubar is possible in the short interval (when both are active) but it not a designed behaviour.
![image](https://github.com/user-attachments/assets/74ad439a-8c29-4cf1-ba70-daecd699bc22)

**Reply by molecularknife** [reply]
this is the design i want to have

**Comment by larrybud2004** [comment]
In the latest version, when both the menubar and tabs are enabled (which is the default install), the menubar does not show, it only has the hamburger menu. Not sure what other option is causing this vs the way yours looks. Turning off tabs will show the menubar.

So what other options do you have enabled/disabled to allow you to show both at the same time?

**Reply by GitHubRulesOK** [reply]
@larrybud2004 
Both at the same time is not supported.
It is an unstable state you could manually or programmatically repeatedly achieve .
but as said it is un-intended temporary state between the 2 single options.

**Reply by tuaris** [reply]
> Turning off tabs will show the menubar.

This is exactly what I want:  No tabs and use a standard menu bar.  This is much more visually appealing and functional.

**Comment by larsen0815** [comment]
> switch off tabs and restart SumatraPDF the windows will be un-tabbed so the mix of tabs and menubar is possible in the short interval (when both are active) but it not a designed behaviour. ![image](https://private-user-images.githubusercontent.com/12754913/418374632-74ad439a-8c29-4cf1-ba70-daecd699bc22.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEyOTI2NDAsIm5iZiI6MTc0MTI5MjM0MCwicGF0aCI6Ii8xMjc1NDkxMy80MTgzNzQ2MzItNzRhZDQzOWEtOGMyOS00Y2YxLWJhNzAtZGFlY2Q2OTliYzIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA2VDIwMTkwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliNzBjM2I1N2M2NmEzOTk3ZDVmOTVkOGVjZTI2Y2NhNTRmZTg4YzM3YzQwMGI3NjlmNTQ0OTkyMTQ4ODU2MTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.X7r9_YAYzluls2rKfYdoWZBTzPqlNf4VX9Q8RVAG-z4)

I'd also like to have only a single window with multiple tabs, and the window having a "normal" menu bar instead of the burger button. Just the way it looks in your screenshot above would be fine. 


> Both at the same time is not supported. It is an unstable state you could manually or programmatically repeatedly achieve . but as said it is un-intended temporary state between the 2 single options.

A little bit confusing to me: it is "not supported", but at the same time it can be achieved somehow. So, technically it is possible, isn't it? And how?

**Reply by larsen0815** [reply]
Already found the answer in another discussion:

> it happens when you start without tabs then switch them tabs on

So, technically **it is possible**. I think many users including me would really like this.
Are there any reasons for not implementing it as a persistent feature?

**Reply by larrybud2004** [reply]
Doesn't work for me. If I have the full menu on and then turn on tabs, the tabs don't appear. But if I then close the program and reopen it, then the tabs show, but the menu is the hamburger.

**Reply by larsen0815** [reply]
Try this:
* Setting "tabs" is off
* Restart SumatraPDF and open one PDF (Menu is shown)
* Set "tabs" on
* Open another PDF
* Both PDFs are shown in separate tabs in one window
